File: sleepy-test/python-unittest/AsyncHelperTests_ContinuationTimeoutNotHitTest.py
import unittest
import time

class TestAsyncHelper(unittest.TestCase):
    def test_continuation_timeout_not_hit(self):
        exceptions = []

        cont = AsyncHelpers.with_timeout(
            AsyncHelpers.prevent_multiple_calls(exceptions.append),
            0.05
        )

        cont(None)
        time.sleep(0.1)

        self.assertEqual(1, len(exceptions))
        self.assertIsNone(exceptions[0])

        cont(None)
        cont(Exception("Some exception"))
        cont(None)
        cont(Exception("Some exception"))

        self.assertEqual(1, len(exceptions))
        self.assertIsNone(exceptions[0])

File: refactored/duplicate-assert/java-junit/AsyncHelperTests_ContinuationTimeoutNotHitTest.java
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertNull;

public class AsyncHelperTests {

    @Test
    public void continuationTimeoutNotHitTest1() {
        {
            List<Exception> exceptions = new ArrayList<>();
            var cont = AsyncHelpers.withTimeout(AsyncHelpers.preventMultipleCalls(exceptions::add), 50);
            cont.accept(null);
            try {
                Thread.sleep(100);
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
            assertEquals(1, exceptions.size());
            assertNull(exceptions.get(0));
        }
    }

    @Test
    public void continuationTimeoutNotHitTest2() {
        {
            List<Exception> exceptions = new ArrayList<>();
            var cont = AsyncHelpers.withTimeout(AsyncHelpers.preventMultipleCalls(exceptions::add), 50);
            cont.accept(null);
            try {
                Thread.sleep(100);
            } catch (InterruptedException e) {
                Thread.currentThread().interrupt();
            }
            cont.accept(null);
            cont.accept(new ApplicationException("Some exception"));
            cont.accept(null);
            cont.accept(new ApplicationException("Some exception"));
            assertEquals(1, exceptions.size());
            assertNull(exceptions.get(0));
        }
    }
}
